### ISSUE TITLE
BigIntegerInstantiation false positive with new BigDecimal( 2 )

### ISSUE THREAD

**Issue by esfomeado** [pmd/pmd#4996]
**Affects PMD Version:** 
7.1.0

**Rule:**

Please provide the rule name and a link to the rule documentation:
https://docs.pmd-code.org/pmd-doc-7.1.0/pmd_rules_java_performance.html#bigintegerinstantiation

**Description:**

**Code Sample demonstrating the issue:**

```java

public interface EvaluationValue 
{
    static EvaluationValue of( BigDecimal value )
    {
        return new NumberEvaluationValue( value );
    }
}

public record NumberEvaluationValue( BigDecimal value )
    implements EvaluationValue
{
    public NumberEvaluationValue( BigDecimal value )
    {
        this.value = value ;
    }
}

...
EvaluationValue.of( new BigDecimal( 2 ) );
...

```

**Expected outcome:**

PMD reports a violation on the `new BigDecimal( 2 )` because it thinks it can be replaced with `BigInteger.TWO` which is not possible.

**Running PMD through:** *Gradle*

**Comment by jsotuyod** [comment]
@esfomeado [`BigDecimal.TWO`](https://docs.oracle.com/en/java/javase/19/docs/api/java.base/java/math/BigDecimal.html#TWO) exists and is valid starting from Java 19. The rule is aware of that and reports it accordingly when the Java version in use allows for it.

So, I'm a little confused as to why you say "it thinks it can be replaced with BigInteger.TWO **which is not possible**.".

Your project is either using Java 19+ and the replacement should be possible, or your Gradle project is misconfigured, and is running PMD for a Java version that doesn't match your actual program.

**Comment by jsotuyod** [comment]
Yes, it seems Gradle is never actually configuring the language version when running PMD, so it always defaults to latest.

Fixing this lies on Gradle however. I've found this issue https://github.com/gradle/gradle/issues/21519 where the lack of proper language version configuration (and therefore preview features) are mentioned.

**Comment by esfomeado** [comment]
I will close this as it's indeed an issue with Gradle.
